test/Loop/DriverTest.php

```diff
@@ -106,6 +107,52 @@ public function testLoopRunsCanBeConsecutiveAndNested()
         });
     }
 
+    public function testCorrectTimeoutIfBlockingBeforeActivate()
+    {
+        $start = 0;
+        $invoked = 0;
+
+        $this->start(function (Driver $loop) use (&$start, &$invoked) {
+            $loop->defer(function () use ($loop, &$start, &$invoked) {
+                $start = getCurrentTime();
+
+                $loop->delay(1000, function () use (&$invoked) {
+                    $invoked = getCurrentTime();
+                });
+
+                \usleep(500000);
+            });
+        });
+
+        $this->assertNotSame(0, $start);
+        $this->assertNotSame(0, $invoked);
+
+        $this->assertGreaterThan(1000, $invoked - $start);
+        $this->assertLessThan(1100, $invoked - $start);
```

Comment:
This is currently expected to fail, because watchers are only activated in the next tick.